containers/src/Kokkos_ScatterView.hpp

```diff
@@ -199,35 +211,30 @@ struct ScatterValue;
 
 template <typename ValueType, typename DeviceType>
 struct ScatterValue<ValueType, Kokkos::Experimental::ScatterSum, DeviceType,
-                    Kokkos::Experimental::ScatterNonAtomic>
-    : Sum<ValueType, DeviceType> {
+                    Kokkos::Experimental::ScatterNonAtomic> {
+  ValueType& value;
+
  public:
   KOKKOS_FORCEINLINE_FUNCTION ScatterValue(ValueType& value_in)
-      : Sum<ValueType, DeviceType>(value_in) {}
+      : value(value_in) {}
   KOKKOS_FORCEINLINE_FUNCTION ScatterValue(ScatterValue&& other)
-      : Sum<ValueType, DeviceType>(other.reference()) {}
+      : value(other.value) {}
```

Comment:
Why do we provide a move constructor in the first place?

Reply:
I didn't want to mess with it here. But yeah I didn't know either really.

Reply:
So as to disable the copy constructor?

Reply:
With the introduction of the `value` data member that is a reference, the type is definitely not copyable nor assignable now :/